### PR TITLE
RI: Updated xpath for phone number collection

### PR DESCRIPTION
**Pull request by linzjax**
Not sure if this is the cleanest way, but all the other information is being pull from a spreadsheet that didn't contain phone numbers, and individual bio pages weren't guaranteed to contain phone numbers.

### CODE REVIEW

File: openstates/ri/legislators.py [hunk 1]
```diff
@@ -98,6 +92,19 @@ def scrape(self, chamber, term):
                 "town_represented": d['town_represented'],
             }
 
+            if chamber == 'upper':
+                contact_url = 'http://webserver.rilin.state.ri.us/Email/SenEmailListDistrict.asp'            
+            if chamber == 'lower':
+                contact_url = 'http://webserver.rilin.state.ri.us/Email/RepEmailListDistrict.asp'
+            contact = lxml.html.fromstring(self.get(contact_url).text)
```
Comment (rshorey) [inline openstates/ri/legislators.py]: self.lxmlize can be used instead here. It's a function we wrote (you can see the code here if you're interested: https://github.com/sunlightlabs/openstates/blob/master/openstates/utils/lxmlize.py)

**Comment by rshorey** [comment]
It looks like the phone number is being added to both the legislator's office (which we want) and as a top-level field in the legislator's JSON (which we don't). Can you make it stop being added to the top level, please?

File: openstates/ri/legislators.py [hunk 2]
```diff
@@ -98,6 +92,19 @@ def scrape(self, chamber, term):
                 "town_represented": d['town_represented'],
             }
 
+            if chamber == 'upper':
+                contact_url = 'http://webserver.rilin.state.ri.us/Email/SenEmailListDistrict.asp'            
+            if chamber == 'lower':
+                contact_url = 'http://webserver.rilin.state.ri.us/Email/RepEmailListDistrict.asp'
```
Comment (rshorey) [inline openstates/ri/legislators.py]: Can you also add the contact URL to the source for the legislator please?

**Comment by rshorey** [comment]
A few legislators' phone numbers seem to be missing. For example, Anastasia Williams' number is available here: http://webserver.rilin.state.ri.us/Email/RepEmailListDistrict.asp but isn't being scraped. I think it may have to do with her having a middle initial in one source but not the other. Maybe it would make sense to match on district instead of first name on the phone list instead (although who knows what problems that creates). I'm OK with missing a few if there's no good way to do it, just wanted to check.

**Comment by linzjax** [comment]
Yeah I noticed a few homepage urls aren't being pull because of differences in how names appear "Cynthia Armour Coyne" vs "Cynthia A. Coyne".
It's also not pulling Da Ponte's number because he's listed as DaPonte on the contact sheet.

I'll poke around for a bit.

**Comment by rshorey** [comment]
this is a super irritating problem. I hate when state legislatures make us match two different pages to each other based on inconsistent names or something. Let us know what you come up with, thanks!

**Comment by linzjax** [comment]
homepage_url feels a little sloppy, but William W. O'Brien was the only one not working with this new method because of a typo in his name : "William W.O'Brien", which is why he has his own code.

**Comment by linzjax** [comment]
Ugh... Frank Lombardo III... hang on...

**Comment by mileswwatkins** [comment]
Lil' Rhody!!

**Comment by linzjax** [comment]
Hah! Take that Frank Lombardo III....

**Comment by rshorey** [comment]
Great job with the Frank-slaying! One more thing - this one's a tiny bit nitpicky (sorry), but since phone is no longer on the top level, we've got a bit of unused code we should remove to prevent later confusion. In line 55, we can make the whole thing an empty dictionary:

d = {}

and we can get rid of the check to see if phone exists (lines 122-123). After that, ready to go!

**Comment by mileswwatkins** [comment]
I was about to say that she could refer to the OS API documents, but they're apparently they don't note which fields are deprecated and which aren't. Still, a good source of information otherwise:
https://sunlightlabs.github.io/openstates-api/

**Comment by linzjax** [comment]
Done! Let me know if there's anything else I should update.

**Comment by rshorey** [comment]
Looks good!

**Pull request merged**